File: core/src/main/java/org/selyu/smp/core/scoreboard/CoreScoreboardAdapter.java
package org.selyu.smp.core.scoreboard;

import org.jetbrains.annotations.NotNull;
import org.selyu.ui.scoreboard.adapter.ScoreboardAdapter;
import org.selyu.ui.scoreboard.objective.ScoreboardObjective;
import org.selyu.ui.scoreboard.title.ScoreboardTitle;

import java.util.List;

import static org.selyu.smp.core.util.MessageUtil.rainbow;

public final class CoreScoreboardAdapter implements ScoreboardAdapter {
    @Override
    public @NotNull ScoreboardTitle getTitle() {
        return ScoreboardTitle.of(false, 1, rainbow("SMP"));
    }

    @Override
    public @NotNull List<ScoreboardObjective> getObjectives() {
        return ScoreboardObjective
                .builder()
                .addObjective("&cHi!", "&bHi!", "&aHi!", "&eHi!")
                .build();
    }
}

File: core/src/main/kotlin/org/selyu/smp/core/scoreboard/CoreScoreboardAdapter.kt
package org.selyu.smp.core.scoreboard

import org.selyu.smp.core.util.rainbow
import org.selyu.ui.scoreboard.adapter.ScoreboardAdapter
import org.selyu.ui.scoreboard.objective.ScoreboardObjective
import org.selyu.ui.scoreboard.title.ScoreboardTitle

class CoreScoreboardAdapter : ScoreboardAdapter {
    override fun getObjectives(): MutableList<ScoreboardObjective> =
            ScoreboardObjective
                    .builder()
                    .addObjective("&dWelcome to the SMP!")
                    .build()

    override fun getTitle(): ScoreboardTitle = ScoreboardTitle.of(false, "Selyu SMP".rainbow()).also {
        it.updateSpeed = 1
    }
}